### PR TITLE
refactor: Fix errors after installing widget to react/next.js apps

### PR DESCRIPTION
**Pull request by igorosip0v**
[WID-181](https://linear.app/worldcoin/issue/WID-181/fix-the-errors-after-installing-idkit-in-react-app)

- Fix problems with Next.js SSR and React on import
- Add examples folders to test widget easier
- Add `eslint`
- Fix `eslint` errors
- Improve build/dev
- ~Remove `zustand` (There was problems with Next.js SSR). Replace it with `StoreContext`. It's working almost the same way. Same syntax when using it.~
- Exclude `cjs` from esbuild. React & Next.js working fine now with `zustand`
- Update README

### CODE REVIEW

**Comment by m1guelpf** [comment]
@igorosip0v Is there any way we can keep Zustand? I'm using it in #5 to allow doing `IDKit.open()` on the Vanilla side, but that won't work with context since we're outside the lifecycle.

**Comment by michaelfurutin** [comment]
@m1guelpf Zustand doesn't work on server side nextjs. Maybe we could implement the opening of the widget using an DOM-event?

**Comment by m1guelpf** [comment]
> @m1guelpf Zustand doesn't work on server side nextjs. Maybe we could implement the opening of the widget using an DOM-event?

Do we really need SSR at all? The modal will always be closed on first render, so it's not like there's anything to SSR either? IMO we could just force rendering on the client and be done with it.

**Comment by michaelfurutin** [comment]
> Do we really need SSR at all? The modal will always be closed on first render, so it's not like there's anything to SSR either? IMO we could just force rendering on the client and be done with it.

Third party developers will have to include the widget in their applications using a dynamic mechanism with server rendering disabled. It is not comfortable.

**Comment by igorosip0v** [comment]
@m1guelpf @paolodamico @michaelfurutin 
Please, take a look now. I think it will be easier to pull the branch and check locally. It seems like everything is working. There was a problem in the cjs build. Esbuild wraps import with `__toESM()` function that adds wrapper object around of zustand import, so it looks like 
```js
{
  default: {
    default: ...,
    ...
  }
  ...
}
  ```
  But after the build import call looks like `var create = require('zustand').default`. In this case `default` will be just an object. And when the code below tries to invoke `create` an error happens, because `create` is not a function.

File: example-nextjs/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import Head from "next/head";
+import "@worldcoin/idkit/build/index.css";
```
Comment (m1guelpf) [inline example-nextjs/pages/index.tsx:2]: This shouldn't be required, as the styles get loaded automatically by the widget.

File: example-react/src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import "@worldcoin/idkit/build/index.css";
```
Comment (m1guelpf) [inline example-react/src/App.tsx:1]: Same as above, this import shouldn't be needed.

**Review by m1guelpf — APPROVED** [review]
LGTM, just left some notes

**Review by m1guelpf — APPROVED** [review]

**Pull request merged**